src/recommendationservice/recommendation_server.py

```diff
@@ -32,26 +36,31 @@
 from grpc_health.v1 import health_pb2
 from grpc_health.v1 import health_pb2_grpc
 
-from metrics import (
-    init_metrics
-)
+application = newrelic.agent.register_application(timeout=10.0)
+
+# from metrics import (
+#     init_metrics
+# )
 
 cached_ids = []
 first_run = True
 
 class RecommendationService(demo_pb2_grpc.RecommendationServiceServicer):
+
+
     def ListRecommendations(self, request, context):
         prod_list = get_product_list(request.product_ids)
-        span = trace.get_current_span()
-        span.set_attribute("app.products_recommended.count", len(prod_list))
+        # span = trace.get_current_span()
+        # span.set_attribute("app.products_recommended.count", len(prod_list))
+        newrelic.agent.add_custom_attribute("products_recommended_count", len(prod_list))
         logger.info(f"Receive ListRecommendations for product ids:{prod_list}")
 
         # build and return response
         response = demo_pb2.ListRecommendationsResponse()
         response.product_ids.extend(prod_list)
 
         # Collect metrics for this service
-        rec_svc_metrics["app_recommendations_counter"].add(len(prod_list), {'recommendation.type': 'catalog'})
+        # rec_svc_metrics["app_recommendations_counter"].add(len(prod_list), {'recommendation.type': 'catalog'})
```

Comment:
This is another opportunity to show the difference between otel and NR agents. Two ideas

1. You could simply record an equivalent timeslice metric using the NR python agent API, or
2. You could use the the OTel metric SDK in conjunction with the NR agent just to record this metric.